File: CaseStudies/FileInputTests/Bundled/Design2/main.jl
# Bundled

# Struct for holding parameters.
struct Parameters
    name::AbstractString
    height::Integer
    weight::AbstractFloat

    # This needs to be an inner constructor now, as it has
    # the same type signature as the struct itself.
    function Parameters(name::AbstractString, height::Integer, weight::AbstractFloat)
    
        parameters = new(name, height, weight)
        input_constraints(parameters) # This needs to be specified as "after initialization" or something
        return parameters
    end
end

function input_constraints(p::Parameters)

    if isempty(strip(p.name))
        throw("Name must not be blank!")
    elseif p.height <= 0
        throw("Height must be positive!")
    elseif p.weight <= 0
        throw("Weight must be positive!")
    end
end

# No longer part of the Parameters 'class'
function get_input(filename::AbstractString)

    file = open(filename, "r")
    lines = readlines(file)
    close(file)

    name = lines[2]
    height = parse(Int32, lines[4])
    weight = parse(Float64, lines[6])

    return Parameters(name, height, weight)
end

p = get_input("FileInputTests/input.txt")
println(p)

File: CaseStudies/FileInputTests/Bundled/Design2/main.swift
// Bundled

import Foundation

// Class for holding parameters
class Parameters {
    let name: String // Default values not needed now!
    let height: Int
    let weight: Double

    init(_ name: String,_ height: Int, _ weight: Double) throws {

        (self.name, self.height, self.weight) = (name, height, weight)
        try self.input_constraints()
    }

    private func input_constraints() throws -> Void {

        if self.name.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            throw ParamError("Name must not be blank!")
        }
        else if self.height <= 0 {
            throw ParamError("Height must be positive!")
        }
        else if self.weight <= 0 {
            throw ParamError("Weight must be positive!")
        }
    }

}

func get_input(_ filename: String) throws -> Parameters {

    let lines = try String(contentsOfFile: filename, encoding: String.Encoding.utf8)
        .components(separatedBy: "\n")
    
    let name: String = lines[1]
    let height: Int = Int(lines[3]) ?? 0
    let weight: Double = Double(lines[5]) ?? 0

    return try Parameters(name, height, weight)
}

// Simple error struct
struct ParamError: LocalizedError {
    let description: String

    init(_ description: String) {
        self.description = description
    }

    var errorDescription: String? {
        description
    }
}

var p: Parameters = try get_input("FileInputTests/input.txt")
print("Parameters(\"" + p.name + "\", " + String(p.height) + ", " + String(p.weight) + ")")